### PR TITLE
WT-12979 Update test/format to take a config when opening a session

### PR DESCRIPTION
**Pull request by etienneptl**
- `wt_wrap_open_session` can now take a custom session configuration
- Preliminary work to unblock WT-12978

### CODE REVIEW

File: test/format/format_util.c [hunk 1]
```diff
@@ -386,13 +386,14 @@ atou32(const char *tag, const char *s, int match)
...
 {
     WT_SESSION *session;
 
     *sessionp = NULL;
 
-    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+    testutil_check(conn->open_session(conn, NULL, cfg, &session));
 
     if (g.trace_conn != NULL && sap->trace == NULL)
         testutil_check(g.trace_conn->open_session(g.trace_conn, NULL, NULL, &sap->trace));
```
Comment (etienneptl) [inline test/format/format_util.c:399]: @sueloverso, do I need to pass the same session configuration here as well?

Comment (sueloverso) [reply test/format/format_util.c:399]: No, this is the session just for calling `log_printf` on the operations. There is no prefetching or any other special thing we do in that other connection. There is no data written.

**Review by sueloverso — APPROVED** [review]
LGTM assuming the other ticket will use this new functionality soon.

**Comment by etienneptl** [comment]
> This LGTM, other than the fact that no caller sends in anything other than NULL. I see it is to unblock another ticket which presumably will do that? If so that's fine. (This work could have been done in that one ticket as necessary to send in some config for that ticket.)

Yes, @sueloverso, the prefetch ticket that needs it is coming right after. I wanted to separate the two pieces of work in case the coming prefetch work gets reverted, I still think we should keep that test/format update.

**Pull request merged**